Makefile

```diff
@@ -38,6 +65,22 @@ AGENT_PACKAGE_PLATFORMS ?= $(GOOS)/$(GOARCH)
 # OS_ARCH.
 OCI_IMAGE_PLATFORMS ?= linux/$(GOARCH)
 
+# Build tags passed to go install/build.
+# Example: BUILD_TAGS="minimal provider_kubernetes"
+BUILD_TAGS ?=
+
+# GIT_COMMIT the current git commit of this repository
+GIT_COMMIT ?= $(shell git -C $(PROJECT_DIR) rev-parse HEAD 2>/dev/null)
+
+
+# Build flag passed to go -mod defaults to readonly to support go workspaces.
+# CI should set this to vendor
+JUJU_GOMOD_MODE ?= readonly
```

Comment:
This was changed to "readonly" from "mod"? For workspaces?

Reply:
yep for this reason.